ci/purge_submitted_questionnaire_state.yaml

```diff
@@ -0,0 +1,44 @@
+platform: linux
+image_resource:
+    type: docker-image
+    source:
+        repository: gcr.io/google.com/cloudsdktool/cloud-sdk
+params:
+    SERVICE_ACCOUNT_JSON: ((gcp.service_account_json))
+    PROJECT_ID:
+    DATAFLOW_TEMPLATE_VERSION: '2021-11-12-00_RC00'
+    SUBMITTED_AT_PURGE_OFFSET_IN_SECONDS: 3600
+run:
+    path: bash
+    args:
+        - -exc
+        - |
+            export GOOGLE_APPLICATION_CREDENTIALS=/root/gcloud-service-key.json
+            cat >$GOOGLE_APPLICATION_CREDENTIALS <<EOL
+            $SERVICE_ACCOUNT_JSON
+            EOL
+
+            gcloud auth activate-service-account --key-file "${GOOGLE_APPLICATION_CREDENTIALS}"
+            gcloud config set project "${PROJECT_ID}"
+
+            DATAFLOW_JOB=$(gcloud dataflow jobs run purge_submitted_questionnaire_state \
+            --region europe-west2 \
+            --gcs-location gs://dataflow-templates/"${DATAFLOW_TEMPLATE_VERSION}"/Firestore_to_Firestore_Delete \
+            --parameters \
+            firestoreReadGqlQuery="SELECT * FROM \`questionnaire-state\` WHERE submitted_at < $((`date --utc +%s` - SUBMITTED_AT_PURGE_OFFSET_IN_SECONDS)) AND submitted_at > NULL",\
+            firestoreReadProjectId="${PROJECT_ID}",\
+            firestoreDeleteProjectId="${PROJECT_ID}" \
+            --format="value(id.scope())")
+
+            DATAFLOW_JOB_STATE=$(gcloud dataflow jobs describe "${DATAFLOW_JOB}" --region europe-west2 --format="value(currentState)")
+
+            while [[ $DATAFLOW_JOB_STATE == "JOB_STATE_RUNNING" || $DATAFLOW_JOB_STATE == "JOB_STATE_PENDING" ]]; do
+              sleep 60
+              DATAFLOW_JOB_STATE=$(gcloud dataflow jobs describe "${DATAFLOW_JOB}" --region europe-west2 --format="value(currentState)")
+            done
+
+            if [[ "${DATAFLOW_JOB_STATE}" == "JOB_STATE_DONE" ]]; then
+              echo "Expired questionnaire states purged successfully from Datastore"
```

Comment:
Might be worth updating the message.

Reply:
Are you suggesting Firestore rather than Datastore? As I couldn't workout what "Cloud Firestore in Datastore mode" means in GCP

Reply:
Nah, just from `Expired` to maybe like `Submitted questionnaire states ...` etc as the expired pipeline has the same message atm.

Reply:
Oh yeah, missed that.